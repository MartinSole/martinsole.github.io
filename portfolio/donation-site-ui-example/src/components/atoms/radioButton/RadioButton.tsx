import React, { Fragment } from 'react'

interface Props {
  handleOnChange?: React.ChangeEventHandler
  id: string
  index: number
  name: string
  text: string
}

const RadioButton = ({ id, index, name, text,handleOnChange }: Props) => {
  return (
    <Fragment>
      <input id={id} name={name} value={text} type="radio" onChange={handleOnChange} defaultChecked={index === 0}/>
      <label htmlFor={id} className='buttonBase'>
        {text}
      </label>
    </Fragment>
  )
}

export default RadioButton