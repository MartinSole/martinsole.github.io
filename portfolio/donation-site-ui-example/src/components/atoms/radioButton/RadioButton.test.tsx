import '@testing-library/jest-dom'
import { fireEvent, render } from "@testing-library/react"
import RadioButton from "./RadioButton"

test("Renders the RadioButton component", () => {
    const mockProps = {
        id: 'cost',
        name: 'cost',
        text: 'Â£5',
        index: 1,
        handleOnChange: jest.fn()
    }
   const {container, getByLabelText} = render(<RadioButton {...{...mockProps}} />)
    expect(container).toMatchSnapshot()
    expect(getByLabelText(mockProps.text)).toBeTruthy
})
