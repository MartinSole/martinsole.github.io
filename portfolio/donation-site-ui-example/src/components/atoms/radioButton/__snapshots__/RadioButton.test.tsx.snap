// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the RadioButton component 1`] = `
<div>
  <input
    id="cost"
    name="cost"
    type="radio"
    value="£5"
  />
  <label
    class="buttonBase"
    for="cost"
  >
    £5
  </label>
</div>
`;
